### PR TITLE
Fix Build Issues on macOS

### PR DESCRIPTION
**Pull request by SConaway**
This PR addresses build issues in macOS. 

First, the include directories of a system `libelf` needs to use the 
`${LIBELF_INCLUDE_DIR}` variable, not just `${LIBELF_INCLUDE}`. 

Second, the `libelf` installed using homebrew doesn't define `EM_RISCV`:
```
/src/machine/programloader.cpp:62:32: error: use of undeclared identifier 'EM_RISCV'
    if (this->hdr.e_machine != EM_RISCV) {
```
It seems to be too old to include it.

### CODE REVIEW

**Comment by jdupak** [comment]
Hi @SConaway,
thanks for raising the issue and suggesting sollution. There is one more issue we need to address: when libelf is installed (as dependency for something else) but it is unusable for us. It seems the best way would be to use https://cmake.org/cmake/help/latest/module/CheckSymbolExists.html to check RISCV compatibility.

If you have time, feel free to look into it. Otherwise I will try to find some time to add that myself.

Anyway, thanks.

File: CMakeLists.txt [hunk 1]
```diff
@@ -212,7 +212,7 @@ else ()
             # Turn non-cmake library into a cmake target
```
Comment (jdupak) [inline CMakeLists.txt:212]: Check of RISC-V symbol in the header here.

**Review by jdupak — CHANGES_REQUESTED** [review]

**Comment by SConaway** [comment]
Yeah, I'll take care of that today or tomorrow. That'll solve the macOS build issue with `libelf` from `brew`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -209,11 +209,18 @@ else ()
     if (NOT "${FORCE_ELFLIB_STATIC}")
         find_package(LibElf)
         if ("${LibElf_FOUND}")
-            # Turn non-cmake library into a cmake target
-            add_library(libelf INTERFACE)
-            target_link_libraries(libelf INTERFACE ${LIBELF_LIBRARY})
-            target_include_directories(libelf INTERFACE ${LIBELF_INCLUDE})
-            message(STATUS "Using system libelf")
+            include(CheckSymbolExists)
```
Comment (SConaway) [inline CMakeLists.txt:212]: Feel free to move this elsewhere

File: CMakeLists.txt [hunk 3]
```diff
@@ -209,11 +209,18 @@ else ()
...
+            check_symbol_exists(EM_RISCV "gelf.h" LIBELF_HAS_RISCV)
+            if ("${LIBELF_HAS_RISCV}")
+                # Turn non-cmake library into a cmake target
+                add_library(libelf INTERFACE)
+                target_link_libraries(libelf INTERFACE ${LIBELF_LIBRARY})
+                target_include_directories(libelf INTERFACE ${LIBELF_INCLUDE_DIR})
+                message(STATUS "Using system libelf")
+            else ()
+                message(STATUS "System libelf does not support RISC-V")
+                set(LibElf_FOUND FALSE) # Force fallback
```
Comment (SConaway) [inline CMakeLists.txt:222]: this way we can treat this as no system libelf and reuse the existing one

**Comment by SConaway** [comment]
I've tested this on both my mac and on Fedora with `elfutils-devel` installed.

**Review by jdupak — APPROVED** [review]
LGTM Thank you

**Pull request merged**